dnf5/main.cpp

```diff
@@ -153,34 +155,33 @@ void RootCommand::set_argument_parser() {
         R"**(Override a configuration option from the configuration file. To override configuration options for repositories, use repoid.option for  the
               <option>.  Values  for configuration options like excludepkgs, includepkgs, installonlypkgs and tsflags are appended to the original value,
               they do not override it. However, specifying an empty value (e.g. --setopt=tsflags=) will clear the option.)**");
-    setopt->set_parse_hook_func([&ctx](
-                                    [[maybe_unused]] ArgumentParser::NamedArg * arg,
-                                    [[maybe_unused]] const char * option,
-                                    const char * value) {
-        auto val = strchr(value + 1, '=');
-        if (!val) {
-            throw std::runtime_error(fmt::format("setopt: Badly formatted argument value \"{}\"", value));
-        }
-        auto key = std::string(value, val);
-        auto dot_pos = key.rfind('.');
-        if (dot_pos != std::string::npos) {
-            if (dot_pos == key.size() - 1) {
-                throw std::runtime_error(
-                    std::string("setopt: Badly formatted argument value: Last key character cannot be '.': ") + value);
+    setopt->set_parse_hook_func(
+        [&ctx](
+            [[maybe_unused]] ArgumentParser::NamedArg * arg, [[maybe_unused]] const char * option, const char * value) {
+            auto val = strchr(value + 1, '=');
+            if (!val) {
+                throw libdnf::cli::ArgumentParserError(M_("setopt: Badly formatted argument value \"{}\""), value);
```

Comment:
Not a blocker for this PR - just thinking whether it makes sense to have a specialized `ArgumentParserInvalidValueError` class for such cases.